File: src/test/resources/mega/mega.cpp
#include "mega.h"

void encode(int n, int l[][3]) {
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < 3; j++) {
			l[i][j] *= 2;
		}
	}
	send(n, l);
}

void decode(int n, int l[]) {
	for (int i = 0; i < n; i++) {
		l[i] *= 2;
	}
	output(n, l);
}

double solve(short a, int b, long long c, float d, double e) {
	return a + b + c + d + e;
}

File: src/test/resources/mega/mega.py
#!/usr/bin/python

"""The Python contestant code for mega."""

import Main

def encode(num_rows, matrix):
    """Encodes the matrix."""
    for i in range(num_rows):
        for j in range(num_rows):
            matrix[i][j] *= 2
    Main.send(num_rows, matrix)

def decode(num_rows, encoded):
    """Decodes the matrix."""
    for i in range(num_rows):
        encoded[i] *= 2
    Main.output(num_rows, encoded)

def solve(num_a, num_b, num_c, num_d, num_e):
    """Sums a bunch of numbers."""
    return num_a + num_b + num_c + num_d + num_e

# vim: set ts=4 sw=4 et :
